### DISCUSSION TITLE
NextJS 11 + SVGR + Typescript

### DISCUSSION THREAD

**Discussion by SalahAdDin** [vercel/next.js#26400] [Help]
Hello guys,

I'm migrating my portfolio from **JavaScript** to **TypeScript** and from **NextJS**  `10` to `11`.

But I'm getting a problem with every **SVG** icon i'm importing:
`TS2604: JSX element type 'GitLab' does not have any construct or call signatures.`.

I followed a lot of tutorials, i tried many ways but i can't fix this; i realized the problem is the global configuration, since when i do click on the importe **SVG** in **Visual Studio Code** it always send me to the **Next** typescript definitions:
`node_modules/next/types/global.d.ts`
![Record_code_20210620233515](https://user-images.githubusercontent.com/5034215/122687645-38e12e00-d220-11eb-9dab-992baf8adc9c.gif)

I deduced the way to fix it is replacing the **SVG** module definition by the custom one:
```typescript
declare module "*.svg" {
  import type { ReactElement, SVGProps } from "react";

  const content: (props: SVGProps<SVGElement>) => ReactElement;
  export default content;
}
```

It is now in a `types/custom.d.ts` and I think there is not a problem importing it because i have another types and there is no problem with them; i deduce the problem is the `global.d.ts` included on **Next**. It does not matter if i declare a `indux.d.ts` at root project or inside of a `@types` folder on the root project, it still does not work.

What's the correct way to override that definition?

Thanks

**Comment by jan-wilhelm** [comment]
dealing with the exact same issue right now, have you found a fix yet?

**Reply by SalahAdDin** [reply]
> dealing with the exact same issue right now, have you found a fix yet?

I handled it by:

1. Setting up a folder layout like the next: 
![image](https://user-images.githubusercontent.com/5034215/128692827-2e14e2fd-a3b2-4735-b0fd-6083d4bf1a9f.png)
2. Declaring a **SVG** type as follows:
```typescript
declare module "*.svg" {
  import type { ReactElement, SVGProps } from "react";

  const content: (props: SVGProps<SVGElement>) => ReactElement;
  export default content;
}
```
![image](https://user-images.githubusercontent.com/5034215/128693033-f737a353-d9dc-44c7-9231-2057a0270fb2.png)
